### ISSUE TITLE
✅ 🔗 🐞 Bug Report: Many many warnings in Dev Console (Duplicate of #19456) - [ED-10226]

### ISSUE THREAD

**Issue by tamelander** [elementor/elementor#22159]
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

### Description

When editing an Elementor page I get a lot of warnings, mostly related to /wp-content/plugins/elementor/assets/js/dev-tools.min.js?ver=3.12.1.
`$e.modules.document.CommandHistory` is hard deprecated since 3.7.0 - Use `$e.modules.editor.document.CommandHistoryBase` instead

`elementor.helpers.getUniqueID()` is hard deprecated since 3.0.0 - Use `elementorCommon.helpers.getUniqueId()` instead

`elementor.helpers.getUniqueID()` is hard deprecated since 3.0.0 - Use `elementorCommon.helpers.getUniqueId()` instead
`container.findChildrenRecursive( callback )` is hard deprecated since 3.5.0 - Use `container.children.findRecursive( callback )` instead

`children` is hard deprecated since 3.0.0 - Use `container.repeaters[ repeaterName ].children` instead

`container.findChildrenRecursive( callback )` is hard deprecated since 3.5.0 - Use `container.children.findRecursive( callback )` instead

I have deactivated all plugins sans
Elementor - 3.12.1	By Elementor.com	
Elementor Pro - 3.12.2	By Elementor.com	
Really Simple SSL - 6.2.0	By Really Simple Plugins

### Steps to reproduce

Editing any page with Elementor. The number of warnings seems to be correlated to the number of widgets on a page. I get these warnings on other sites as well. On one site with large pages containing numerous widgets, the warnings can number in the hundreds. I know they are just warnings, but perhaps the time has come to fix them. I have the same warnings in Elementor 3.11.1.

### Isolating the problem

- [X] This bug happens with only Elementor plugin active (and Elementor Pro).
- [X] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [X] I can reproduce this bug consistently following the steps above.

### System Info

<details>
<summary>System Info</summary>

````txt


					
== Server Environment ==
	Operating System: Linux
	Software: LiteSpeed
	MySQL version: MariaDB Server v10.5.19
	PHP Version: 8.1.17
	PHP Memory Limit: 256M
	PHP Max Input Vars: 3000
	PHP Max Post Size: 512M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.2
	Site URL: https://www.mysite.se/
	Home URL: https://www.mysite.se/
	WP Multisite: No
	Max Upload Size: 256 MB
	Memory limit: 40M
	Max Memory limit: 1024M
	Permalink Structure: /%postname%/
	Language: en-US
	Timezone: 2
	Admin Email: xxx
	Debug Mode: Inactive

== Theme ==
	Name: Hello Elementor
	Version: 2.7.1
	Author: Elementor Team
	Child Theme: No

== User ==
	Role: administrator
	WP Profile lang: sv_SE
	User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/112.0.0.0 Safari/537.36

== Active Plugins ==
	Elementor
		Version: 3.12.1
		Author: Elementor.com

	Elementor Pro
		Version: 3.12.2
		Author: Elementor.com

	Really Simple SSL
		Version: 6.2.0
		Author: Really Simple Plugins


== Must-Use Plugins ==
	vaa-cs.php
		Version: 
		Author: 


== Features ==
	Custom Fonts: 0
	Custom Icons: 0

== Integrations ==
	


== Elementor Experiments ==
	Optimized DOM Output: Active
	Improved Asset Loading: Active
	Improved CSS Loading: Active
	Inline Font Icons: Inactive
	Accessibility Improvements: Active by default
	Additional Custom Breakpoints: Active by default
	admin_menu_rearrangement: Inactive by default
	Flexbox Container: Active
	Upgrade Swiper Library: Inactive by default
	Default to New Theme Builder: Active by default
	Hello Theme Header & Footer: Inactive by default
	Editor Top Bar: Inactive by default
	Landing Pages: Inactive
	Nested Elements: Inactive by default
	Lazy Load Background Images: Inactive by default
	Page Transitions: Inactive
	Notes: Active by default
	Loop: Active by default
	Form Submissions: Active by default
	Scroll Snap: Active by default
	Menu: Inactive by default


== Log ==
	


== Elementor - Compatibility Tag ==
	
	Elementor Pro: Incompatible

== Elementor Pro - Compatibility Tag ==
	

				
````

</details>

**Comment by nicholaszein** [comment]
Hello @tamelander! 

Thank you for submitting your issue! 🙏 

This is a known issue, and we are currently working on a fix. Right now we don't have an ETA on the problem, but this shouldn't affect the functionality and workflow of the builder. So there is nothing you should be worried about.

Best regards

**Comment by HypeNeutron** [comment]
> Hello @tamelander!
> 
> Thank you for submitting your issue! 🙏
> 
> This is a known issue, and we are currently working on a fix. Right now we don't have an ETA on the problem, but this shouldn't affect the functionality and workflow of the builder. So there is nothing you should be worried about.
> 
> Best regards

@nicholaszein
**Actually** affect the functionality and workflow of the builder
When I use Xpro Addons for elementor , I can't edit container I also using Element pro and Hello theme that problem make my work at freeze! 🤦

**Comment by hirschferkel** [comment]
I get a 20 seconds delay before I can edit my pages with elementor since the last Update to 3.17.3 with notice from Firefox: This site slkows down yoour browser.

`children` is hard deprecated since 3.0.0 - Use `container.repeaters[ repeaterName ].children` instead
`container.findChildrenRecursive( callback )` is hard deprecated since 3.5.0 - Use `container.children.findRecursive( callback )` instead

![Bildschirmfoto 2023-11-25 um 15 42 17](https://github.com/elementor/elementor/assets/16016422/96c5be6b-15e6-4c62-9e29-08aa38a1d9e3)

**Comment by hirschferkel** [comment]
I get a 20 seconds delay and complete Firefox freeze before I can edit my pages on OS X Monterey with elementor since the Update from 3.15 up ... with notice from Firefox (Safe Mode): This site slows down your browser speed.
Chrome will work as expected.

`children` is hard deprecated since 3.0.0 - Use `container.repeaters[ repeaterName ].children` instead
`container.findChildrenRecursive( callback )` is hard deprecated since 3.5.0 - Use `container.children.findRecursive( callback )` instead

![Bildschirmfoto 2023-11-25 um 15 42 17](https://github.com/elementor/elementor/assets/16016422/96c5be6b-15e6-4c62-9e29-08aa38a1d9e3)
![Bildschirmfoto 2023-11-25 um 16 07 26](https://github.com/elementor/elementor/assets/16016422/9da74ca3-52c5-4619-9134-2cfc8fc07ce3)
![Bildschirmfoto 2023-11-25 um 16 06 32](https://github.com/elementor/elementor/assets/16016422/58451f0e-f043-4be1-bd12-bf43a80dbf4b)

**Comment by tamelander** [comment]
As stated in the OP @hirschferkel, these are warnings and might not be related with your slow loading problems at all. I have turned off warnings in Chrome.
Use the Chrome dev console's Network panel and see if anything seems amiss. 
Do you get 20 sec wait on a small Elementor page as well as a large Elementor page?
I have started to regularly clean the database. 
Disabled or put in learning mode Wordfence firewall.
Deleted page revisions. 
Emptied the database wp_actionscheduler_actions and wp_actionscheduler_logs tables.

**Comment by hirschferkel** [comment]
@tamelander I get it on every Elementor page. But as I mentioned only in Firefox and not in Chrome. So the database cannot be the source of the lag, could it?
Chrome just loads the UI ...

**Comment by tamelander** [comment]
>>So the database cannot be the source of the lag, could it?
Yes indeed it can. I removed over 30 MB of wp_actions_sheduler data. 
If you are diligent with making backups of you database (easily done in Cpanel for instance) you can decrease the number of post revisions that is saved. I have turned them off completely. 
Make a new temp database, copy your existing database to it, and change to the test database i wp-config.php. Make sure that both databases works. Set your site to use the temp database. Install this plugin (no affiliation) and let it do its work. It cleans and removes orphaned meta data amongst other things. My database size was halfed.
https://wordpress.org/plugins/database-cleaner/

Firefox quite recently added some form of strict script handling. I don't remember what it's called, google the warnings you get in Firefox and you'll learn.   

I have a hosting with 2048M of php RAM. I allow Wordpress to use 1024M.  You need at least 512M if I recall correctly. Compare what you got in your hosting php ram with the memory size of your page, as: In Google Chrome, after your page is loaded, you can hover your curser over the Chrome window (or tab) and it will tell you how much memory the page uses.

**Comment by hirschferkel** [comment]
@tamelander thanks for you considerations. 
I googled it before I came here, but there was no clue about Elementor ... The page uses only 144MB with elementor enabled and RAM is 2048M.
I use an automatic deletion of page reviews of 5 reviews per page.
Currently I hesitate to do the things you mentioned, as it takes really some time to figure that out. My databse is currently very small, so ... 
So far I just use Chrome, now. Maybe if I have some time next day, I will check out your steps ...

**Comment by tamelander** [comment]
I don't want to insult your intelligence, but you have disabled all forms of caching? 
I usually work in Google Chrome Incognito window with the dev console open and [ x ] Disable cache checked.

The Google Chrome Dev console network panel
and perhaps Query monitor plugin would be my steps.
Cheers

**Comment by hirschferkel** [comment]
Caching is disabled, sure. ;D
O.k. I'll check it out next weeks, best.

**Comment by hirschferkel** [comment]
@tamelander It was a bug: https://github.com/elementor/elementor/issues/24432#issuecomment-1829919931

**Comment by tamelander** [comment]
@hirschferkel Praise the lard! Sorry for leading you astray. Only wanted to help.

**Comment by hirschferkel** [comment]
@tamelander no prob! You just tried to figure it out. So It was just to letting you know about the issue being fixed.

**Comment by nicholaszein** [comment]
Hello everyone!

### 📢 We're happy to announce that the issue you raised was resolved in Elementor Core v3.19.0! 🥳 

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
